keras/src/models/model.py

```diff
@@ -156,6 +156,15 @@ def __init__(self, *args, **kwargs):
             functional.Functional.__init__(self, *args, **kwargs)
         else:
             Layer.__init__(self, *args, **kwargs)
+            if args:
+                inputs = args[0]
```

Comment:
This assumes there is only one input argument. That's true for Functional models, that is not necessarily true for subclass model.